posts/tree-cell-typing/index.qmd

```diff
@@ -0,0 +1,134 @@
+
+* `cth`:  a nested list object that describes the relationship between cell types, or the **c**ell **t**ype **h**ierarchy
+* `quantile_absolute_expression_difference_param` and `quantile_percent_expression_difference_param`: numbers that define how differentiating
+each gene must be to be included in the supervised cell type approach
+
+# InSituTree Example: Running the package {#sec-example-implementation}
+
+You can find the InSituTree package [here](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/InSituTree){target="_blank"}. See the corresponding [vignette](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/InSituTree/vignettes/InSituTree_basicVignette.html){target="_blank"} inside the package for more details. 
+
+For this example we work with the immune-focused cell type profiles 
+as well as the mini non-small cell lung cancer dataset included
+in the InSituType package. We start by setting up a nested list of cell types.
+```{r setupCellTypeTree, message=FALSE, eval = FALSE}
+# Read in io profiles
+data(ioprofiles)
+
+# Establish a nested list of cell types
+celltype_hierarchy_list <- list(
+  "structural" = c(
+    "endothelial", 
+    "fibroblast"),
+  "myeloid" = c(
+    "macrophage", 
+    "mast", 
+    "mDC", 
+    "monocyte", 
+    "neutrophil"),
+  "lymphoid" = list(
+    "B-lymphoid" = c(
+      "B-cell", 
+      "pDC",
+      "plasmablast"),
+    "T-lymphoid" = list(
+      "T4" = c(
+        "T CD4 memory", 
+        "T CD4 naive", 
+        "Treg"),
+      "T8" = c(
+        "T CD8 memory", 
+        "T CD8 naive"),
+      "NK")
+    )
+)
+
+
+```
+
+:::{.callout-tip}
+If you're working from a CosMx-derived cell profile from our [CosMx Cell Profile library](https://github.com/Nanostring-Biostats/CosMx-Cell-Profiles/tree/main){target="_blank"}, formatted nested cell type lists are available as `.celltypeslist.R` files.
+:::
+
+Once we have this cell type tree, we can use InSituTree to run supervised cell typing with a nested approach.
+```{r runInSituTree, message=FALSE, eval = FALSE}
+# Load basic data consisting of:
+#   - counts: cell x gene count matrix
+#   - negmeans: vector of mean negative probe counts per cell
+
+load("mini_nsclc")
+
+# Run InSituTree with basic options
+res <- InSituTree(
+  x = as(mini_nsclc$counts, "CsparseMatrix"),
+  neg = rowMeans(mini_nsclc$neg),
+  full_profiles = ioprofiles,
+  cth = celltype_hierarchy_list,
+  name_for_new_annotation = "nested_cell_type",
+  cohort = NULL,
+  excluded_genes = NULL,
+  quantile_absolute_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in absolute expression differences
+  quantile_percent_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in percentage expression differences
+  return_summary_annotation = TRUE
+)
+
+# Access the final summarized cell typing results
+res$summaryAnnotation
```

Comment:
Head of `res$summaryAnnotation` is now included as an asset and shown in the post.

Reply:
Great. For completeness of this post, please add a brief sentence to say that final cell typing results at different layers are included in separate columns and the right most 2 columns are for final results.

Reply:
Added. This section now reads: 
This summary annotation results data frame contains cell typing results at each level of annotation in separate columns, with both the cell type annotation and the posterior probability of that assignment *among the cell types considered at a given branch*. The final, most resolved, cell typing results will be found in the rightmost two columns.
Cells which have no counts among the selected genes for the first level of annotations will appear with `NA` values
for all columns. Otherwise, final annotations will be cascaded right in the summary table.